chirpstack/src/api/device.rs

```diff
@@ -538,6 +538,13 @@ impl DeviceService for Device {
                 .map_err(|e| e.status())?;
         }
 
+        // LoRaWAN 1.1 devices send a mac-command when changing to Class-C. Change the class here for LoRaWAN 1.0 devices.
+        if dp.supports_class_c && dp.mac_version.to_string().starts_with("1.0") {
+            let _ = device::set_enabled_class(&dev_eui, DeviceClass::C).await;
```

Comment:
Could you add trailing `.map_err(|e| e.status())?` to both `set_enabled_class` calls to make sure that errors are returned?

Reply:
Done.